home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/CooperationContactServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.softserveinc.ita.homeproject.homeservice.service.impl;
+
+import java.util.List;
+
+import com.softserveinc.ita.homeproject.homedata.entity.Contact;
+import com.softserveinc.ita.homeproject.homedata.entity.ContactType;
+import com.softserveinc.ita.homeproject.homedata.entity.Cooperation;
+import com.softserveinc.ita.homeproject.homedata.repository.ContactRepository;
+import com.softserveinc.ita.homeproject.homeservice.dto.ContactDto;
+import com.softserveinc.ita.homeproject.homeservice.exception.AlreadyExistHomeException;
+import com.softserveinc.ita.homeproject.homeservice.exception.NotFoundHomeException;
+import com.softserveinc.ita.homeproject.homeservice.mapper.ServiceMapper;
+import com.softserveinc.ita.homeproject.homeservice.service.CooperationContactService;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class CooperationContactServiceImpl extends BaseContactService implements CooperationContactService {
+
+    private final CooperationServiceImpl cooperationService;
+
+    public CooperationContactServiceImpl(ContactRepository contactRepository,
+                                         ServiceMapper mapper,
+                                         CooperationServiceImpl cooperationService) {
+        super(contactRepository, mapper);
+        this.cooperationService = cooperationService;
+    }
+
+    @Override
+    protected void checkAndFillParentEntity(ContactDto contactDto, Contact createContact, Long parentEntityId) {
+        var cooperation = mapper
```

Comment:
Shouldn't we check that cooperation is enabled?

Reply:
in method getCooperationById (in Cooperation service) we check for enabled or not